internal/lifecycle/readme_test.go

```diff
@@ -0,0 +1,438 @@
+package lifecycle
+
+import (
+	"bufio"
+	"errors"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func setEnv(e map[string]string) error {
+	for k, v := range e {
+		err := os.Setenv(k, v)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func unsetEnv(e map[string]string) error {
+	for k, _ := range e {
+		err := os.Unsetenv(k)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func equalError(a, b error) bool {
+	if a == nil {
+		return b == nil
+	}
+	if b == nil {
+		return a == nil
+	}
+	return errors.Is(a, b)
+}
+
+type test struct {
+	in                   string
+	codeBlocks           []codeBlock
+	cmds                 []*exec.Cmd
+	toCommandsErr        error
+	extractLifecycleErr  error
+	extractCodeBlocksErr error
+	env                  map[string]string
+	serviceName          string
+	gcrURL               string
+}
+
+var tests = []test{
+	{
+		in: "[//]: # ({sst-run-unix})\n" +
```

Comment:
Consider multiline strings or a single line - this feels a bit hard to read.

Reply:
Resolved offline, thank you.

Reply:
For reference: we're keeping this because it doesn't seem line multiline strings support escaped backticks.